### PR TITLE
Fix warnings and edit-and-continue

### PR DESCRIPTION
**Pull request by Flamefire**
Enable /ZI
Dynamic link of lua
Rebuild and add bzip2d.lib/pdb
Fix MSVC resource compile
Make files to copy a list (required till CMake 3.5)
Disable SafeSEH for debug

### CODE REVIEW

File: cmake/Modules/CMakeMacroAddFlags.cmake [hunk 1]
```diff
@@ -7,8 +7,8 @@ MACRO(ADD_FLAGS parameter)
 	# Make a copy of the current arguments in ${parameter}
 	SET(new_parameter ${${parameter}})
 	
-        # fix arch-argument bug
-        STRING(REPLACE "-arch;" "-arch=" new_parameter "${new_parameter}")
```
Comment (Flow86) [inline cmake/Modules/CMakeMacroAddFlags.cmake]: is the indentation change intentional?

File: cmake/Modules/CMakeMacroAddFlags.cmake [hunk 2]
```diff
@@ -37,8 +37,8 @@ MACRO(ADD_FLAGS parameter)
 		ENDFOREACH(option ${TMP})
 	ENDFOREACH(required_arg ${ARGN})
 	
-        # unfix arch-argument bug
-        STRING(REPLACE "-arch=" "-arch " new_parameter "${new_parameter}")
```
Comment (Flow86) [inline cmake/Modules/CMakeMacroAddFlags.cmake]: is the indentation change intentional?

File: cmake/Modules/CMakeMacroForceAddFlags.cmake [hunk 3]
```diff
@@ -37,8 +37,8 @@ MACRO(FORCE_ADD_FLAGS parameter)
 		ENDFOREACH(option ${TMP})
 	ENDFOREACH(required_arg ${ARGN})
 	
-        # unfix arch-argument bug
-        STRING(REPLACE "-arch=" "-arch " new_parameter "${new_parameter}")
+    # unfix arch-argument bug
+    STRING(REPLACE "-arch=" "-arch " new_parameter "${new_parameter}")
```
Comment (Flow86) [inline cmake/Modules/CMakeMacroForceAddFlags.cmake]: is the indentation change intentional?

File: cmake/Modules/CMakeMacroAddFlags.cmake (hunk 1 above)
Comment (Flamefire) [reply cmake/Modules/CMakeMacroAddFlags.cmake]: Yes. It was one level to much (same for the others)

**Pull request merged**